src/modules/launcher/PowerLauncher/Storage/QueryHistory.cs

```diff
@@ -41,5 +41,21 @@ public void Add(string query)
                 });
             }
         }
+
+        public void Update()
+        {
+            foreach (var item in Items)
+            {
+                if (item.ExecutedDateTime == DateTime.MinValue)
+                {
+                    item.ExecutedDateTime = DateTime.Now;
+                }
+
+                if (string.IsNullOrEmpty(item.Query))
```

Comment:
I'm not sure if this makes sense 🤔 I'm fine with overwriting execution time, but if Query is malformed and we couldn't load it properly, overwriting it with "Default query" doesn't make it usable in the future. I'd suggest if Query is null or empty, just remove that history entry

Reply:
let's do this - `if Query is null or empty, just remove that specific history entry`